### PR TITLE
DjangoCon 2014: extract speakers info

### PR DESCRIPTION
**Pull request by redapple**
Related to https://github.com/pyvideo/pyvideo-data/pull/31#issuecomment-217929137

This is the script I used:

```
import re

from clive import lib


data = lib.load_json_data('data/djangocon-2014/videos')

for path, info in data:
    summary = info.get('summary', '')
    lines = summary.splitlines()
    first_line = lines[0]
    m = re.compile(r'^By, (.+)$').search(first_line)
    if m:
        info['speakers'] = [s.strip() for s in m.group(1).split(',')]
        info['summary'] = "\n".join(lines[1:]).strip()

lib.save_json_data(data)
```

### CODE REVIEW

**Comment by redapple** [comment]
"Various presenters" is debatable: https://github.com/pyvideo/pyvideo-data/pull/107/files#diff-9721f22d0b627ae2a838fecf58c434c8R22

File: data/djangocon-2014/videos/anatomy-of-a-django-project.json [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-  "description": "",
-  "quality_notes": "",
-  "language": "English",
-  "copyright_text": "",
-  "thumbnail_url": "https://i.ytimg.com/vi/ajEDo1semzs/maxresdefault.jpg",
-  "duration": 1290,
+  "category": "DjangoCon 2014", 
+  "slug": "anatomy-of-a-django-project", 
+  "title": "Anatomy of a Django Project", 
+  "summary": "By Mark Lavin\nWebsites built with Django are built on \"projects\" which are composed of oneor more \"apps\". But what is a project really? This talk will dissect a Django project to understand which pieces are convention and which are required. It explain what if anything separates a project from an app and answer common questions about projects vs apps.\n\nHelp us caption & translate this video!\n\nhttp://amara.org/v/FNEk/",
```
Comment (willkg) [inline data/djangocon-2014/videos/anatomy-of-a-django-project.json]: Missed this one.

File: data/djangocon-2014/videos/a-nice-problem-to-have-django-under-heavy-load.json [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
+  "thumbnail_url": "https://i.ytimg.com/vi/gukzap2BU2M/maxresdefault.jpg", 
+  "duration": 2547, 
   "videos": [
     {
-      "type": "youtube",
-      "length": 0,
-      "url": "https://www.youtube.com/watch?v=gukzap2BU2M"
+      "url": "https://www.youtube.com/watch?v=gukzap2BU2M", 
+      "length": 0, 
+      "type": "youtube"
```
Comment (willkg) [inline data/djangocon-2014/videos/a-nice-problem-to-have-django-under-heavy-load.json:16]: I'm kind of surprised these are changing order and that this PR seems to add whitespace to the end of lines. That's a bug in save/load_json functions, I think. They should round-trip files exactly.

File: data/djangocon-2014/videos/choose-your-own-django-deployment-adventure.json [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
     }
-  ],
-  "source_url": "https://www.youtube.com/watch?v=QrFEKghISEI",
-  "tags": [],
-  "speakers": [],
+  ], 
+  "source_url": "https://www.youtube.com/watch?v=QrFEKghISEI", 
+  "tags": [], 
+  "speakers": [
+    "Kate Heddleston & Matt Makai"
```
Comment (willkg) [inline data/djangocon-2014/videos/choose-your-own-django-deployment-adventure.json]: This should get split into two separate speakers.

File: data/djangocon-2014/videos/geo-django-geo-beyond-the-django.json [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
-  "description": "",
-  "quality_notes": "",
-  "language": "English",
-  "copyright_text": "",
-  "thumbnail_url": "https://i.ytimg.com/vi/mUhinowr3RY/maxresdefault.jpg",
-  "duration": 1466,
+  "category": "DjangoCon 2014", 
+  "slug": "geo-django-geo-beyond-the-django", 
+  "title": "Geo+Django: Geo beyond the Django", 
+  "summary": "B, Joe Jasinski\nHave you gone through the comprehensive GeoDjango docs, but wondered where to go next? Are you curious about how you can combine the power GeoDjango with other community-built tools? Do you want to create pretty maps in Python? If so, you are in the right place. Learn about GeoDjango and Geographic Information Systems and navigate beyond the docs into the exciting GIS technology landscape.\n\nHelp us caption & translate this video!\n\nhttp://amara.org/v/FOPU/",
```
Comment (willkg) [inline data/djangocon-2014/videos/geo-django-geo-beyond-the-django.json]: Missed this one, too.

File: data/djangocon-2014/videos/high-performance-django-from-runserver-to-reddit.json [hunk 5]
```diff
@@ -1,23 +1,23 @@
...
-  "description": "",
-  "quality_notes": "",
-  "language": "English",
-  "copyright_text": "",
-  "thumbnail_url": "https://i.ytimg.com/vi/Toa9lW8UMOA/maxresdefault.jpg",
-  "duration": 2364,
+  "category": "DjangoCon 2014", 
+  "slug": "high-performance-django-from-runserver-to-reddit", 
+  "title": "High Performance Django: From Runserver to Reddit Hugs", 
+  "summary": "By Peter Baumgartner\nDjango makes it easy to build a site and get it running on your laptop, but how do you go from there to a site that can gracefully handle millions of page views per day? This talk will show you the modifications and supporting services needed to make your site scale. Topics will include caching, uWSGI, Varnish, and load balancing.\n\nHelp us caption & translate this video!\n\nhttp://amara.org/v/FPWi/",
```
Comment (willkg) [inline data/djangocon-2014/videos/high-performance-django-from-runserver-to-reddit.json]: Missed this one.

File: data/djangocon-2014/videos/lightning-talks.json [hunk 6]
```diff
@@ -1,23 +1,25 @@
...
     }
-  ],
-  "source_url": "https://www.youtube.com/watch?v=NMwuzJP4Z_o",
-  "tags": [],
-  "speakers": [],
+  ], 
+  "source_url": "https://www.youtube.com/watch?v=NMwuzJP4Z_o", 
+  "tags": [], 
+  "speakers": [
+    "Various Presenters"
```
Comment (willkg) [inline data/djangocon-2014/videos/lightning-talks.json]: I'd nix this--it's not a real person. For lightning talks, I typically figure out who all the speakers are and then list them individually. We should continue doing that, but we can write up an issue and then do it in a future PR.

File: data/djangocon-2014/videos/technical-onboarding-training-and-mentoring.json [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
-  "description": "",
-  "quality_notes": "",
-  "language": "English",
-  "copyright_text": "",
-  "thumbnail_url": "https://i.ytimg.com/vi/B1dZ3AjRJJI/maxresdefault.jpg",
-  "duration": 1704,
+  "category": "DjangoCon 2014", 
+  "slug": "technical-onboarding-training-and-mentoring", 
+  "title": "Technical Onboarding, Training, and Mentoring", 
+  "summary": "By Kate Heddleston\nWith the increase of code academies training new engineers there is an increase in junior engineers on the market. A lot of companies are hesitant to hire too many young engineers because they lack sufficient resources to train them.\n\nHelp us caption & translate this video!\n\nhttp://amara.org/v/FPWe/",
```
Comment (willkg) [inline data/djangocon-2014/videos/technical-onboarding-training-and-mentoring.json]: Missed this one.

**Comment by willkg** [comment]
That script did a great pass. There's some things I'd like to change. If you have the time and inclination, it'd be great if you could fix them and then push another commit to this branch.

If not, we can merge this as is and I can fix the issues I had.

Whichever way works for you.

Thank you for doing this!

**Comment by redapple** [comment]
@willkg , I believe I fixed the speakers issues you raised.
About whitespace added, indeed, weird.

**Comment by willkg** [comment]
This is awesome--thank you!

**Comment by willkg** [comment]
@redapple Pretend I didn't check... does this mean we need to run the md -> reST script and redo that PR? If so, I can do that later today.

**Comment by redapple** [comment]
Yes, I believe so. Should go smoothly. Sorry, I could not take care of this
today
Le 10 mai 2016 7:24 PM, "Will Kahn-Greene" notifications@github.com a
écrit :

**Pull request merged**